hal/src/main/native/athena/FRCDriverStation.cpp

```diff
@@ -531,7 +387,9 @@ static void newDataOccur(uint32_t refNum) {
   // Since we could get other values, require our specific handle
   // to signal our threads
   if (refNum != refNumber) return;
-  dsThread->Notify();
+  // Notify all threads
+  newDSDataAvailableCounter++;
```

Comment:
@ThadHouse Is there a particular reason that this increment isn't protected by `newDSDataAvailableMutex`?

Reply:
Thats a bug. Fixed in #2034